services/app-api/src/resolvers/healthPlanPackage/unlockHealthPlanPackage.test.ts

```diff
@@ -468,6 +468,91 @@ describe('unlockHealthPlanPackage', () => {
         )
     })
 
+    it('send state email to state contacts and all submitters when unlocking submission succeeds', async () => {
+        const config = testEmailConfig
+        const mockEmailer = testEmailer(config)
+        //mock invoke email submit lambda
+        const stateServer = await constructTestPostgresServer()
+        const stateServerTwo = await constructTestPostgresServer({
+            context: {
+                user: {
+                    id: 'PeterParker',
+                    stateCode: 'FL',
+                    role: 'STATE_USER',
+                    email: 'notspiderman@example.com',
+                    familyName: 'Parker',
+                    givenName: 'Peter',
+                },
+            },
+        })
+
+        // First, create a new submitted submission
+        const stateSubmission = await createAndSubmitTestHealthPlanPackage(
+            stateServer
+        )
+
+        const cmsServer = await constructTestPostgresServer({
+            context: {
+                user: testCMSUser,
+            },
+            emailer: mockEmailer,
+        })
+
+        // First unlock
+        await unlockTestHealthPlanPackage(
+            cmsServer,
+            stateSubmission.id,
+            'Super duper good reason.'
+        )
+
+        // Resubmission to get multiple submitters
+        await resubmitTestHealthPlanPackage(
+            stateServerTwo,
+            stateSubmission.id,
+            'Test resubmission reason'
+        )
+
+        // Final unlock to test against
+        const unlockResult = await unlockTestHealthPlanPackage(
+            cmsServer,
+            stateSubmission.id,
+            'Super duper good reason.'
+        )
+
+        const currentRevision = unlockResult.revisions[0].node.formDataProto
+
+        const sub = base64ToDomain(currentRevision)
+        if (sub instanceof Error) {
+            throw sub
+        }
+
+        const programs = [defaultFloridaProgram()]
+        const ratePrograms = [defaultFloridaRateProgram()]
+        const name = packageName(sub, programs)
+        const rateName = generateRateName(sub, sub.rateInfos[0], ratePrograms)
+
+        const stateReceiverEmails = [
+            'james@example.com',
+            'notspiderman@example.com',
+            ...sub.stateContacts.map((contact) => contact.email),
+        ]
+
+        // email subject line is correct for CMS email.
+        // Mock emailer is called 4 times, 2 for the first unlock, 2 for the second unlock.
+        // From the pair of emails, the second one is the state email.
+        expect(mockEmailer.sendEmail).toHaveBeenNthCalledWith(
```

Comment:
I like that you are checking sendEmail and able to confirm the toAddresses directly, rather than just intercepting the call to sendUnlockPackage… 

I’m always a bit hestiant to use “toHaveBeenCalledWith” but I think this makes a lot of sense. I wonder if we could bulid out a test emailer that you could pull the raw email texts out of and assert agasint those instead.

Reply:
> I wonder if we could bulid out a test emailer that you could pull the raw email texts out of and assert agasint those instead.

That would be an interesting idea.

> I’m always a bit hestiant to use “toHaveBeenCalledWith” ...

Could you elaborate more on this?

I'll merge this now so we can get the fix in, but please feel free to respond.